SmartDeviceLink/private/SDLMenuConfigurationUpdateOperation.m

```diff
@@ -0,0 +1,99 @@
+//
+//  SDLMenuConfigurationUpdateOperation.m
+//  SmartDeviceLink
+//
+//  Created by Joel Fischer on 1/21/21.
+//  Copyright © 2021 smartdevicelink. All rights reserved.
+//
+
+#import "SDLMenuConfigurationUpdateOperation.h"
+
+#import "SDLError.h"
+#import "SDLGlobals.h"
+#import "SDLLogMacros.h"
+#import "SDLSetGlobalProperties.h"
+#import "SDLVersion.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SDLMenuConfigurationUpdateOperation ()
+
+@property (weak, nonatomic) id<SDLConnectionManagerType> connectionManager;
+@property (strong, nonatomic) NSArray<SDLMenuLayout> *availableMenuLayouts;
+@property (strong, nonatomic) SDLMenuConfiguration *updatedMenuConfiguration;
+@property (assign, nonatomic) SDLMenuConfigurationUpdatedBlock menuConfigurationUpdatedBlock;
+
+@property (copy, nonatomic, nullable) NSError *internalError;
+
+@end
+
+@implementation SDLMenuConfigurationUpdateOperation
+
+- (instancetype)initWithConnectionManager:(id<SDLConnectionManagerType>)connectionManager windowCapability:(SDLWindowCapability *)windowCapability newMenuConfiguration:(SDLMenuConfiguration *)newConfiguration configurationUpdatedHandler:(SDLMenuConfigurationUpdatedBlock)configurationUpdatedBlock {
+    self = [super init];
+    if (!self) { return nil; }
+
+    _connectionManager = connectionManager;
+    _availableMenuLayouts = windowCapability.menuLayoutsAvailable;
+    _updatedMenuConfiguration = newConfiguration;
+    _menuConfigurationUpdatedBlock = configurationUpdatedBlock;
+
+    return self;
+}
+
+- (void)start {
+    [super start];
+    if (self.isCancelled) { return; }
+
+    if (self.availableMenuLayouts.count == 0) {
+        SDLLogW(@"Could not set the main menu configuration. Which menu layouts can be used is not available");
+        self.internalError = [NSError sdl_menuManager_configurationOperationLayoutsNotSupported];
+        return [self finishOperation];
+    } else if (![self.availableMenuLayouts containsObject:self.updatedMenuConfiguration.mainMenuLayout]
+               || ![self.availableMenuLayouts containsObject:self.updatedMenuConfiguration.defaultSubmenuLayout]) {
```

Comment:
No, that's a public property, so it would be a major version change. Also, they do different things such that `submenuLayout` wouldn't be a good name.